src/Security/Member.php

```diff
@@ -1702,4 +1711,52 @@ public function getHtmlEditorConfigForCMS()
         // If can't find a suitable editor, just default to cms
         return $currentName ? $currentName : 'cms';
     }
+
+    /**
+     * This needs to be public because it's referenced in getMemberPasswordField() and called in ConfirmedPasswordField
+     */
+    public static function generateRandomPassword(array $args = []): string
+    {
+        $password = '';
+        $validator = self::password_validator();
+        $minLength = $validator ? $validator->getMinLength() : 20;
+        $maxLength = $args['maxLength'] ?? 0;
+        if ($maxLength === 0) {
+            $maxLength = max($minLength, 20);
+        } else {
+            if ($maxLength < $minLength) {
+                throw new InvalidArgumentException('maxLength argument is less than PasswordValidator minLength');
+            }
+        }
+        // Create a dummy member to test the password validator against
+        // Use a dummy member because this is intended to be used in a context where an Administrator
+        // is created a branch new member who doesn't yet have an ID
+        $dummyMember = Member::create();
+        // These are similar character sets that used in the default PasswordValidator
+        $charsets = [
+            'abcdefghijklmnopqrstuvwyxz',
+            'ABCDEFGHIJKLMNOPQRSTUVWYXZ',
+            '0123456789',
+            '!@#$%^&*()_+-=[]{};:,./<>?',
+        ];
+        $n = 0;
+        do {
+            $valid = true;
+            $password = '';
+            for ($i = 0; $i < $maxLength; $i++) {
+                $charset = $charsets[$i % 4];
+                $randomInt = random_int(0, strlen($charset) - 1);
+                $password .= $charset[$randomInt];
+            }
+            $password = str_shuffle($password);
```

Comment:
https://www.php.net/manual/en/function.str-shuffle.php
> Caution
>
> This function does not generate cryptographically secure values, and must not be used for cryptographic purposes, or purposes that require returned values to be unguessable.
>
> If cryptographically secure randomness is required, the [Random\Randomizer](https://www.php.net/manual/en/class.random-randomizer.php) may be used with the [Random\Engine\Secure](https://www.php.net/manual/en/class.random-engine-secure.php) engine. For simple use cases, the [random_int()](https://www.php.net/manual/en/function.random-int.php) and [random_bytes()](https://www.php.net/manual/en/function.random-bytes.php) functions provide a convenient and secure API that is backed by the operating system’s CSPRNG.

We should probably not use `str_shuffle` here.

Reply:
We're using random_int() above which is cryptographically secure

Reply:
Any time a password has predictability, it's a bad thing. Without `str_shuffle` it would be known that these passwords have certain types of characters in certain positions, which makes it a lot easier to brute force. The fact that `str_shuffle` isn't cryptographically secure implies that it's _still_ possible to know the what type of characters will be in which positions.

Reply:
This still has not been responded to.

If you're _very_ confident that this is not a problem, I can leave it. But please respond to that effect.

Reply:
Have changed to shuffle the characters using random_int() which is listed as cryptographically secure